examples/memory_with_compression/main.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=wrong-import-position
+"""The main entry point of the MemoryWithCompress example."""
+import asyncio
+import os
+import sys
+from pathlib import Path
+
+# Add the project root to Python path to enable imports
+project_root = Path(__file__).parent.parent.parent
+if str(project_root) not in sys.path:
+    sys.path.insert(0, str(project_root))
+
+# Imports must come after sys.path manipulation
+from agentscope.agent import ReActAgent  # noqa: E402
+from agentscope.formatter import DashScopeChatFormatter  # noqa: E402
+from agentscope.message import Msg  # noqa: E402
+from agentscope.model import DashScopeChatModel  # noqa: E402
+from agentscope.tool import Toolkit  # noqa: E402
+from agentscope.token import OpenAITokenCounter  # noqa: E402
+
+from examples.memory_with_compression._memory_with_compress import (  # noqa: E402, E501
```

Comment:
Just import from  the current dir, e.g. `from _memory_with_compress import MemoryWithCompress`

Reply:
fix: 1. add  compress_on_get and compress_on_add 2. add public method check_and_compress() 3. fix the import problem